.github/workflows/docker_snapshot.yml

```diff
@@ -0,0 +1,81 @@
+name: Build internal snapshot image
+
+on:
+  push:
+    branches:
+      - 'master'
+      - '[0-9]+.[0-9]+.z'
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'hazelcast'
+    steps:
+      - name: Checkout hazelcast-snapshot repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast-dockerfiles/hazelcast-snapshot
+          ref: master
```

Comment:
This may be redundant. Default branch should be master if this isn't the repo that is triggered. https://github.com/actions/checkout#usage.

Reply:
You are right, it would default to it even without the property. Nevertheless, referencing the branch name explicitly could simplify adjustments if we need for some branches to specify a custom branch in the `hazelcast-dockerfiles/hazelcast-snapshot`.